### DISCUSSION TITLE
Is it possible to use Elsa with separate diagramming tool?

### DISCUSSION THREAD

**Discussion by cryo75** [elsa-workflows/elsa-core#707] [Q&A]
I'm looking in implementing workflow management into my app. My app consists of an ASP.Net WebApi2 server and a .Net 4.8 Winforms client.

Currently I do have a diagramming tool implemented, which simply allows me to add shapes and connections and store the info as xml. I would like to keep this tool. Now I need to extend this tool to support workflows.

So I came across Elsa and it looks very interesting. My idea would be to only use Elsa to execute tasks. I would need to map / translate my diagram to Elsa workflows. I will also handle all db persistence. Workflows will be executed client-side for now. Maybe server-side at a later stage.

Is something like this possible to do?

**Comment by sfmskywalker** [comment]
Definitely possible. With Elsa 2, you'll have a REST API at your disposal you can communicate with from your WinForms application. It allows you to list activity types (to e.g. implement an activity picker), and get property descriptors for a given activity (to render e.g. an activity editor).

The process would be very similar to the designer built with StencilJS, which consumes the same API.
